baybe/parameters/fidelity.py

```diff
@@ -0,0 +1,269 @@
+"""Fidelity parameters."""
+
+from functools import cached_property
+from typing import Any, ClassVar, cast
+
+import cattrs
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from attrs.validators import and_, deep_iterable, ge, le, min_len
+from typing_extensions import override
+
+from baybe.parameters.base import DiscreteParameter
+from baybe.parameters.enum import CategoricalEncoding
+from baybe.parameters.validation import (
+    validate_contains_one,
+    validate_is_finite,
+    validate_unique_values,
+)
+from baybe.utils.numerical import DTypeFloatNumpy
+
+
+@define(frozen=True, slots=False)
+class CategoricalFidelityParameter(DiscreteParameter):
+    """Parameter class for categorical fidelity parameters."""
+
+    # class variables
+    is_numerical: ClassVar[bool] = True
+    # See base class.
+
+    # object variables
+    encoding: CategoricalEncoding | None = field(
+        init=False,
+        default=None,
+    )
+    # See base class.
+
+    _values: tuple[str, ...] = field(
+        alias="values",
+        converter=lambda x: cattrs.structure(x, tuple[str, ...]),  # type: ignore
+        validator=[
+            min_len(2),
+            validate_unique_values,  # type: ignore
+        ],
+    )
+    # See base class.
+
+    _costs: tuple[float, ...] = field(
+        alias="costs",
+        # FIXME[typing]: https://github.com/python-attrs/cattrs/issues/111
+        converter=lambda x: cattrs.structure(x, tuple[float, ...]),
+        # FIXME[typing]: https://github.com/python-attrs/attrs/issues/1197
+        validator=[
+            min_len(2),
+            validate_is_finite,
+            deep_iterable(ge(0.0)),
```

Comment:
please use a keyword argument here since it's otherwise unclear to which part the `ge` belongs